chain/src/txhashset.rs

```diff
@@ -659,16 +661,8 @@ impl<'a> Extension<'a> {
 		let commit = out.commitment();
 
 		if let Ok(pos) = self.batch.get_output_pos(&commit) {
-			// we need to check whether the commitment is in the current MMR view
-			// as well as the index doesn't support rewind and is non-authoritative
-			// (non-historical node will have a much smaller one)
-			// note that this doesn't show the commitment *never* existed, just
-			// that this is not an existing unspent commitment right now
-			if let Some(hash) = self.output_pmmr.get_hash(pos) {
-				// Check the hash matches what we expect.
-				// We may be on a fork which may result in the entry at that pos being
-				// different to the one we expect.
-				if hash == OutputIdentifier::from_output(out).hash_with_index(pos - 1) {
+			if let Some(out_mmr) = self.output_pmmr.get_data(pos) {
```

Comment:
As you mention this will hardfork archive nodes because `get_data` bypasses the spent check. We should only check UTXOs.

Reply:
We are only checking UTXOs here.

The variants are `get_data()` and `get_data_from_file()`. 
`get_data()` _is_ "spent aware", the other goes straight to the file.

We should think about some better naming for these two...

Reply:
Oh right, sorry for the confusion!